### ISSUE TITLE
artefacts on thumbnails in lighttable

### ISSUE THREAD

**Issue by mabod** [darktable-org/darktable#10386]
darktable is showing artefacts (green squares) on the thumbnails in the lighttable:

![Bildschirmfoto_2021-11-11_07-29-09](https://user-images.githubusercontent.com/10911665/141249228-a5b357eb-6e69-4e58-9911-3be9b488828d.png)

The view in the darkroom is not affected and the exported pictures are good.

This is with darktable 2:3.6.1-2 on Arch Linux. The OS is up to date. It happens with kernel  5.15.1.arch1-1 and 5.10.78-1. Downgrading darktable to 2:3.6.0-1 did not help.

This is on a Ryzen 7 CPU with with a Radeon RX 5600 XT GPU. The PC is running

amd-ucode 20211027.1d00989-1
opencl-amd 21.30.1290604-1
xf86-video-amdgpu 21.0.0-2

**Comment by TurboGit** [comment]
I would try to disable OpenCL. If this works, then try to update the driver if not up-to-date.

**Comment by mabod** [comment]
I disabled opencl but it does not help

**Comment by TurboGit** [comment]
Hove you regenerated the thumbs? Do you have the same artifacts in the darkroom?

**Comment by mabod** [comment]
I did not regenerate the thumbs. The thumbs are ok. The artefacts are not in the thumbs but generated by darktable on the fly. I see the green squares mvoing for the first 1-2 seconds after startup before they stay as they are.

The darkroom does not show does not show artefacts on the main picture but in the film strip at the bottom.

`darktable --disable-opencl`

![Bildschirmfoto vom 2021-11-12 07-19-15](https://user-images.githubusercontent.com/10911665/141422937-d43ff482-2551-4dba-81ec-2a35f4da0acd.png)

**Comment by AlicVB** [comment]
That's really very strange ??? even more as from 3.4 to 3.6 there has been a full rewrite of the lighttable code (filmtrip use the same code) and how the thumbnails are drawn...

To be absolutely sure, can you verify the thumbnails files and check that they don't have the grreen patterns ? The thumb files should be located in `~/.cache/darktable/mipmaps-.../[0-7]/...`
Another point : does other gtk application show such artefacts ?

**Comment by mabod** [comment]
I checked the thumbnails with thunar (XFCE). They are all fine. I double checked  explicitly those thumbnails which I see in darktable. In thunar they are fine, in darktable they have green squares. 

I havent seen that issue with any other app. e.g not in gimp, not in firefox.

**Comment by mabod** [comment]
And I have to say that this behavior is new. I frequently use darktable. I have used darktable 3.6 before without this issue. This issue must have come up during the last couple of weeks.

**Comment by AlicVB** [comment]
> And I have to say that this behavior is new

I'm confused now... don't you say that it happen also with a 3.4 version ???

**Comment by TurboGit** [comment]
This reminds me of  https://github.com/darktable-org/darktable/issues/3234

But you'll need to provide more information and try to understand what triggers this.

Is that for all images? Only RAW? Just after import? Only when using a specific module? etc...

Again, when such issue was reported it was always OpenCL, so double check that.

Until we can reproduce there is nothing more we can do!

**Comment by mabod** [comment]
> > And I have to say that this behavior is new
> 
> I'm confused now... don't you say that it happen also with a 3.4 version ???

I am sorry. That was a typo. I tested with downgrading to 3.6.0 and that does not help. I corrected the typo in my opening post.

**Comment by mabod** [comment]
> This reminds me of #3234
> 
> But you'll need to provide more information and try to understand what triggers this.
> 
> Is that for all images? Only RAW? Just after import? Only when using a specific module? etc...
> 
> Again, when such issue was reported it was always OpenCL, so double check that.
> 
> Until we can reproduce there is nothing more we can do!

opencl is definitely not the root cause. I disabled it in the GUI and I double checked by starting darktable with `darktable --disable-opencl`

The issue is for all existing thumbnails in lighttable or in the filmstrip. it does not affect the main picture in darktoom and it does not effect the output. 

I happens in lighttable even if I turn off all modules for a specific picture. The new thumbnail then still shows the green squares. And they are moving when I hover the mouse over the thumbnails. 

And it happens within gnome and with XFCE.

And this issue was not there a couple of weeks ago. I certainly never had that issue with DT 3.4. And not with DT 3.6 at the beginning. It must have to do with one of the Arch packages I have updated in the meantime. But I do not know where to look. I keep my Arch uptodate on a daily basis.

**Comment by TurboGit** [comment]
> And this issue was not there a couple of weeks ago. I certainly never had that issue with DT 3.4. And not with DT 3.6 at the beginning.

So that's some OS libs that you have upgraded. Looks at the updates on your machine... Maybe the display driver?

**Comment by TurboGit** [comment]
Anyway at this stage it seems clear that this is not a dt issue.

**Comment by mabod** [comment]
This is certainly an issue with some library that was updated. But so far only DT has issues with it. I am lost finding the right library to downgrade. And I am actually puzzled why I seem to be the only Arch user with that issue.

Anyways, here is a short video showing the issue. It shows how darktable starts with opencl disabled and how the green squares move for the first few seconds after the start. When the squares stop moving, I move the mouse and the screen squares start to change with it.

https://user-images.githubusercontent.com/10911665/141627303-ca249a12-ffcb-4e02-b70e-046a8a44d3a1.mp4

**Comment by AlicVB** [comment]
whaou, nice effect indeed ;)
I really don't see how that can be related to dt code directly. I think it's more that dt is stressing the system more than other app...
For the packages, I would check : 
1. gtk packages (but I'm not sure)
2. graphics drivers (most certainly the issue imo) : as you have an amd card, can you test what happen with the "legacy" driver (not the one provided by amd) I'm not sure how to do that, but iirc that's possible.

some other things : 
- does it happen also in culling ?
- same question for full preview.
- same question for slideshow
- changing the thumbnail size change something ?
- same question if you change the window size

**Comment by TurboGit** [comment]
Could it be LittleCMS related and you have the color managed lightable activated in preferences? Try disabling this option and see if it still reproduces. If not, check if LittleCMS was updated recently.

**Comment by mabod** [comment]
LittleCMS was not updated recently and I do not see a specific lighttable preference for LittleCMS. So I dont know what to check/uncheck here. 


> * does it happen also in culling ?

I dont know what "culling" means in this context. 
 
> * same question for full preview.

When I use "W" or "Y" shortcut the picture is showing fine.

> * same question for slideshow

No issues with slideshow

> * changing the thumbnail size change something ?

The green squares appear regardless of thumbnail size.
 
> * same question if you change the window size

No matter what the window size is, the green squares appear always. full screen (F11), manual resizing, etc.

**Comment by mabod** [comment]
In the meantime I successfully compiled DT 3.4.1 (but with -DUSE_OPENEXR=no;  otherwise configure fails). And this DT 3.4.1 has the same issue. So something was changed on my system which affects DT 3.4.1 and 3.6.1 thumbnail rendering. What could that be?

**Comment by AlicVB** [comment]
ok, so the bad news for you is that I really think it comes from your system : 
- culling and full preview use the exact same codes path than thumbnails (they are "just" big thumbnails) so if some dt code was the culprit we would most probably have seen issues in those modes too...
- 3.4.1 expose the bug too, as well as the entire code used for thumbnails display has been change in 3.6. The only point that has not changed is the thumbnail creation, but that's not the problem as the artifact are moving and the thumbnail files are ok...
- you say that you were using dt without issues until recently...

Now, what can be the problem with your install is a good question ! As said earlier, I would bet that your graphic driver is the problem...

I close this issue in order to tidy up a little github, but that only means that the issue is not from dt code, don't hesitate to answer or request a reopening if needed !
you can also ask a question on pixl.us forum, where people may be more helpful than us to solve system issues...

Thanks for your report

**Comment by mabod** [comment]
I found the issue. It has to do with xorg-server /xorg-server-comon 21.1.1-2. This update came through on 10. November 2021.
Downgrading to 1.20.13-3  resolves the issue.

**Comment by TurboGit** [comment]
@mabod : Congrats for finding the issue. If it gets resolved later in a future  xorg-server-comon version do not hesitate to add this information here. It could help others. TIA.

**Comment by mabod** [comment]
There was lots of debugging going on with xorg-server. Even to the point where I had to do a `git bisect` myself. I identified a big commit as the first bad commit:

https://gitlab.freedesktop.org/xorg/xserver/-/issues/1250#note_1475538

But until today the issue is not fixed. Luckily there is a workaround. Setting `AMD_DEBUG=nodcc`

This prevents any of the artifacts with the latest xorg-server/mesa packages.
